src/Formik.tsx

```diff
@@ -298,37 +299,21 @@ export class Formik<Values = object, ExtraProps = {}> extends React.Component<
 
   handleChange = (
     eventOrPath: string | React.ChangeEvent<any>
-  ): void | ((eventOrTextValue: string | React.ChangeEvent<any>) => void) => {
-    // @todo someone make this less disgusting.
-    //
-    // executeChange is the core of handleChange, we'll use it cache change
-    // handlers like Preact's linkState.
+  ): void | ((eventOrValue: any | React.ChangeEvent<any>) => void) => {
+    // this function actually handles the change
     const executeChange = (
-      eventOrTextValue: string | React.ChangeEvent<any>,
+      eventOrValue: any | React.ChangeEvent<any>,
```

Comment:
Is it any? Or a string?

Reply:
A use case I had in my project, were custom input elements that for-example return a number in onChange or even an array (for a tag input). That's why I change the check, not relying on detecting wether the value is a string, but instead wether the value is an event. This way the default case is that onChange has any kind of value and the special case that it is an event like for default html input elements.

Reply:
Maybe we use unkown then.

Reply:
Sure. I didn’t know that type. Should I change that?

Reply:
yes please